zrml/court/src/lib.rs

```diff
@@ -91,12 +110,69 @@ mod pallet {
     where
         T: Config,
     {
+        // No-one can stake more than BalanceOf::<T>::max(), therefore, this function saturates
+        // arithmetic operations.
+        fn current_required_stake(jurors_num: usize) -> BalanceOf<T> {
+            let jurors_len: BalanceOf<T> = jurors_num.saturated_into();
+            T::StakeWeight::get().saturating_mul(jurors_len)
+        }
+
+        // Simply takes the sum of all hash bytes. Each byte addition is processed using mod 17.
+        fn random_usize() -> usize {
+            // https://github.com/paritytech/substrate/issues/8312
+            let (random_hash, _) = T::Random::random(b"zrml-court");
+            let mut random_usize: usize = 0;
+            for byte in random_hash.as_ref().iter().copied() {
+                // This arithmetic operation will never fail
+                let n = Into::<usize>::into(byte) % 17;
+                random_usize = random_usize.saturating_add(n);
+            }
+            random_usize
+        }
+
         // Retrieves a juror from the storage
         fn juror(account_id: &T::AccountId) -> Result<Juror<BalanceOf<T>>, DispatchError> {
             Jurors::<T>::get(account_id).ok_or_else(|| Error::<T>::JurorDoesNotExists.into())
         }
     }
 
+    impl<T> DisputeApi for Pallet<T>
+    where
+        T: Config,
+    {
+        type AccountId = T::AccountId;
+        type Balance = BalanceOf<T>;
+        type BlockNumber = T::BlockNumber;
+        type Origin = T::Origin;
+        type MarketId = MarketIdOf<T>;
+
+        fn on_dispute<D>(
+            dispute_bond: D,
+            _market_id: Self::MarketId,
+            _outcome: OutcomeReport,
+            who: Self::AccountId,
+        ) -> DispatchResult
+        where
+            D: Fn(usize) -> Self::Balance,
+        {
+            CurrencyOf::<T>::reserve(&who, dispute_bond(1))?;
+            let max_random_jurors = Jurors::<T>::iter().count().max(MAX_RANDOM_JURORS);
+            let random_usize = Self::random_usize();
+            let jurors_len_opt = max_random_jurors.checked_rem(random_usize);
+            let fun = || ArithmeticError::Underflow.into();
+            let _jurors_len = jurors_len_opt.ok_or_else::<DispatchError, _>(fun)?;
+            Ok(())
```

Comment:
We should generate randomness to determine each index of the Juror that is selected. So the actual composition of the jury is random.

Reply:
Done. Take a look at `random_jurors`